File: sprint5/By lessons/lesson3code1/lesson3code1.kt
fun findNode(root: Node?, value: Int): Node? {
    // Если мы пришли в поддерево, а его не существует, значит, нужного элемента в дереве поиска нет
    if (root == null) {
        return null // "Значение {value} не найдено"
    }
    return when {
        value < root.value -> findNode(root.left, value)
        value == root.value -> root
        value > root.value -> findNode(root.right, value)
        else -> null
    }
}

File: sprint5/By lessons/lesson3code1/lesson3code1.py
def find_node(root, value):
  # Если мы пришли в поддерево, а его не существует, значит, нужного элемента в дереве поиска нет
  if root is None: 
    return f"Значение {value} не найдено"
  if value < root.value:
    return find_node(root.left, value)
  if value = root.value:
    return root
  if value > root.value:
    return find_node(root.right, value)
